File: sdk/phoenix-client-java/src/main/java/io/github/vishva_kalhara/phoenix/util/PhoenixExceptionHandler.java
package io.github.vishva_kalhara.phoenix.util;

/**
 * Functional interface for handling {@link PhoenixException} occurrences.
 * <p>
 * This interface allows implementing custom exception-handling logic
 * for errors that arise within the Phoenix client.
 * </p>
 */
public interface PhoenixExceptionHandler {

    /**
     * Handles a {@link PhoenixException}.
     *
     * @param e The exception to be handled.
     */
    void handle(PhoenixException e);
}

File: packages/phoenix-client-cs/Phoenix/Util/PhoenixExceptionHandler.cs
namespace Phoenix.Util
{
    /// <summary>
    /// Delegate for handling <see cref="PhoenixException"/> occurrences.
    /// <para>
    /// This delegate allows implementing custom exception-handling logic
    /// for errors that arise within the Phoenix client.
    /// </para>
    /// </summary>
    /// <param name="e">The exception to be handled.</param>
    public delegate void PhoenixExceptionHandler(PhoenixException e);
}